utils/appliance/__init__.py

```diff
@@ -2097,3 +2135,12 @@ def get_or_create_current_appliance():
     return stack.top
 
 current_appliance = LocalProxy(get_or_create_current_appliance)
+
+
+class CurrentAppliance(object):
+    def __init__(self):
+        self.ca = None
+
+    def __get__(self, instance, owner):
+        self.ca = get_or_create_current_appliance()
```

Comment:
it means after first use it will stay the same value for all uses even for other appliances